src/main/java/decodes/tsdb/algo/AW_AlgorithmBase.java

```diff
@@ -1547,7 +1552,16 @@ protected abstract void doAWTimeSlice()
 	protected abstract void afterTimeSlices()
 		throws DbCompException;
 	
-	public abstract String[] getPropertyNames();
+	public String[] getPropertyNames()
+	{
+		org.opendcs.annotations.PropertySpec[] annos = this.getClass().getAnnotationsByType(org.opendcs.annotations.PropertySpec.class);
+		ArrayList<String> ret = new ArrayList<>();
+		for (org.opendcs.annotations.PropertySpec a: annos)
+		{
+			ret.add(a.name());
+		}
+		return ret.toArray(new String[0]);
+	}
```

Comment:
consider putting this logic into AnnotationHelpers.

Reply:
Thanks, new I missed a spot.